packages/test/test-end-to-end-tests/src/test/serializeAfterFailedAttach.spec.ts

```diff
@@ -0,0 +1,219 @@
+import { IFluidHandle, IRequest } from "@fluidframework/core-interfaces";
+import { describeCompat } from "@fluid-private/test-version-utils";
+import { IDocumentServiceFactory } from "@fluidframework/driver-definitions";
+import type { ISharedMap } from "@fluidframework/map";
+import { wrapObjectAndOverride } from "../mocking.js";
+
+describeCompat(
+	`Serialize After Failure to Attach Container Test`,
+	"NoCompat",
+	(getTestObjectProvider, apis) => {
+		const { SharedMap, SharedString } = apis.dds;
+
+		const codeDetails: IFluidCodeDetails = {
+			package: "detachedContainerTestPackage1",
+			config: {},
+		};
+		const sharedStringId = "ss1Key";
+		const sharedMapId = "sm1Key";
+
+		let provider: ITestObjectProvider;
+		let loader: Loader;
+		let request: IRequest;
+		const loaderContainerTracker = new LoaderContainerTracker();
+
+		async function createDetachedContainerAndGetEntryPoint() {
+			const container: IContainer = await loader.createDetachedContainer(codeDetails);
+			// Get the root dataStore from the detached container.
+			const defaultDataStore = (await container.getEntryPoint()) as TestFluidObject;
+			return {
+				container,
+				defaultDataStore,
+			};
+		}
+
+		function createTestLoader(): Loader {
+			const factory: TestFluidObjectFactory = new TestFluidObjectFactory([
+				[sharedStringId, SharedString.getFactory()],
+				[sharedMapId, SharedMap.getFactory()],
+			]);
+			const codeLoader = new LocalCodeLoader([[codeDetails, factory]], {});
+			const testLoader = new Loader({
+				urlResolver: provider.urlResolver,
+				documentServiceFactory: wrapObjectAndOverride<IDocumentServiceFactory>(
+					provider.documentServiceFactory,
+					{
+						createContainer: () => assert.fail("fail on attach"),
+					},
+				),
+				codeLoader,
+				logger: provider.logger,
+				configProvider: {
+					getRawConfig: (name) =>
+						name === "Fluid.Container.RetryOnAttachFailure" ? true : undefined,
+				},
+			});
+			loaderContainerTracker.add(testLoader);
+			return testLoader;
+		}
+
+		const createPeerDataStore = async (containerRuntime: IContainerRuntimeBase) => {
+			const dataStore = await containerRuntime.createDataStore(["default"]);
+			const peerDataStore = (await dataStore.entryPoint.get()) as ITestFluidObject;
+			return {
+				peerDataStore,
+				peerDataStoreRuntimeChannel: peerDataStore.channel,
+			};
+		};
+
+		beforeEach("createLoader", async function () {
+			provider = getTestObjectProvider();
+			if (
+				semver.compare(provider.driver.version, "0.46.0") === -1 &&
+				(provider.driver.type === "routerlicious" || provider.driver.type === "tinylicious")
+			) {
+				this.skip();
+			}
+			const documentId = createDocumentId();
+			request = provider.driver.createCreateNewRequest(documentId);
+			loader = createTestLoader();
+		});
+
+		afterEach("resetLoaderContainerTracker", () => {
+			loaderContainerTracker.reset();
+		});
+
+		const tests = () => {
```

Comment:
i don't think it is necessary to nest this in a function. usually only done to run tests in multiple different ways